internal/components/builder.go

```diff
@@ -15,19 +15,21 @@ import (
 type ParserOption[ComponentConfigType any] func(*Settings[ComponentConfigType])
 
 type Settings[ComponentConfigType any] struct {
-	protocol        corev1.Protocol
-	appProtocol     *string
-	targetPort      intstr.IntOrString
-	nodePort        int32
-	name            string
-	port            int32
-	defaultRecAddr  string
-	portParser      PortParser[ComponentConfigType]
-	rbacGen         RBACRuleGenerator[ComponentConfigType]
-	livenessGen     ProbeGenerator[ComponentConfigType]
-	readinessGen    ProbeGenerator[ComponentConfigType]
-	defaultsApplier Defaulter[ComponentConfigType]
-	envVarGen       EnvVarGenerator[ComponentConfigType]
+	protocol            corev1.Protocol
+	appProtocol         *string
+	targetPort          intstr.IntOrString
+	nodePort            int32
+	name                string
+	port                int32
+	defaultRecAddr      string
+	portParser          PortParser[ComponentConfigType]
+	clusterRoleRulesGen ClusterRoleRulesGenerator[ComponentConfigType]
+	roleGen             RoleGenerator[ComponentConfigType]
+	roleBindingGen      RoleBindingGenerator[ComponentConfigType]
```

Comment:
do we also need to have a cluster role binding gen?

Reply:
It is not needed because it is not associated with a specific namespace.